src/position-manager/libraries/ConfigPermissionsMap.sol

```diff
@@ -0,0 +1,111 @@
+  /// @param status The new status for the `canSetUsingAsCollateral` permission.
+  /// @return The updated ConfigPermissions.
+  function setCanSetUsingAsCollateral(
+    ConfigPermissions self,
+    bool status
+  ) internal pure returns (ConfigPermissions) {
+    return
+      ConfigPermissions.wrap(
+        _setStatus(ConfigPermissions.unwrap(self), CAN_SET_USING_AS_COLLATERAL_MASK, status)
+      );
+  }
+
+  /// @notice Sets the new status for the `canUpdateUserRiskPremium` permission.
+  /// @param self The current ConfigPermissions.
+  /// @param status The new status for the `canUpdateUserRiskPremium` permission.
+  /// @return The updated ConfigPermissions.
+  function setCanUpdateUserRiskPremium(
+    ConfigPermissions self,
+    bool status
+  ) internal pure returns (ConfigPermissions) {
+    return
+      ConfigPermissions.wrap(
+        _setStatus(ConfigPermissions.unwrap(self), CAN_UPDATE_USER_RISK_PREMIUM_MASK, status)
+      );
+  }
+
+  /// @notice Sets the new status for the `canUpdateUserDynamicConfig` permission.
+  /// @param self The current ConfigPermissions.
+  /// @param status The new status for the `canUpdateUserDynamicConfig` permission.
+  /// @return The updated ConfigPermissions.
+  function setCanUpdateUserDynamicConfig(
+    ConfigPermissions self,
+    bool status
+  ) internal pure returns (ConfigPermissions) {
+    return
+      ConfigPermissions.wrap(
+        _setStatus(ConfigPermissions.unwrap(self), CAN_UPDATE_USER_DYNAMIC_CONFIG_MASK, status)
+      );
+  }
+
+  /// @notice Returns whether the `canSetUsingAsCollateral` permission or global permissions are enabled.
+  /// @param self The current ConfigPermissions.
+  /// @return Whether the `canSetUsingAsCollateral` permission or global permissions are enabled.
+  function canSetUsingAsCollateral(ConfigPermissions self) internal pure returns (bool) {
+    return _getStatus(self, CAN_SET_USING_AS_COLLATERAL_MASK);
+  }
+
+  /// @notice Returns whether the `canUpdateUserRiskPremium` permission or global permissions are enabled.
+  /// @param self The current ConfigPermissions.
+  /// @return Whether the `canUpdateUserRiskPremium` permission or global permissions are enabled
+  function canUpdateUserRiskPremium(ConfigPermissions self) internal pure returns (bool) {
+    return _getStatus(self, CAN_UPDATE_USER_RISK_PREMIUM_MASK);
+  }
+
+  /// @notice Returns whether the `canUpdateUserDynamicConfig` permission or global permissions are enabled.
+  /// @param self The current ConfigPermissions.
+  /// @return Whether the `canUpdateUserDynamicConfig` permission or global permissions are enabled
+  function canUpdateUserDynamicConfig(ConfigPermissions self) internal pure returns (bool) {
+    return _getStatus(self, CAN_UPDATE_USER_DYNAMIC_CONFIG_MASK);
+  }
+
+  /// @notice Sets the new status for the given permission.
+  function _setStatus(uint8 self, uint8 mask, bool status) private pure returns (uint8) {
+    return status ? self | mask : self & ~mask;
```

Comment:
lets see if using xor ternary trick to avoid branching is cheaper here (bc else branch has two ops, `and` & `not`, it might not be cheaper)